react/src/reactTextMask.js

```diff
@@ -25,6 +25,33 @@ export default class MaskedInput extends React.PureComponent {
     this.initTextMask()
   }
 
+  componentDidUpdate(prevProps) {
+    // Getting props affecting value
+    const {value, pipe, mask, guide, placeholderChar, showMask} = this.props
+
+    // Сalculate that settings was changed:
+    // - `pipe` converting to string, to compare function content
+    // - `mask` converting to string, to compare values or function content
+    // - `keepCharPositions` exludes, because it affect only cursor position
+    const settings = {guide, placeholderChar, showMask}
+    const isPipeChanged = typeof pipe === 'function' && typeof prevProps.pipe === 'function' ?
+      pipe.toString() !== prevProps.pipe.toString() :
+      false
```

Comment:
If `prevProps.pipe` or `props.pipe` is `undefined` and the other is `function`, then `isPipeChanged` should be `true`?

Reply:
Yes, you are right. Need to be `true`. I'll make changes

Reply:
Ok :-)

I really appreciate your help.

Will you kindly also provide a  test case that covers this?

Reply:
Yes, sure